### ISSUE TITLE
ISSUE, Calling a exteded function from parent class, doesn't work.

### ISSUE THREAD

**Issue by eriknyk** [goaop/framework#131]
Hi, If I have a class

```
class Greeting
{
    public hello($name)
    {
        echo "hello $name";
    }
}
```

And I extend this class

```
class Example extends Greeting
{
}
```

And I have defined on my aspect class:  

```
@Before("execution(public Example->*(*))")
```

And I execute a Example instance.

```
$e = new Example();
$e->hello("erik");
```

I doesn't work, the aspect class is not matching the class Example->*.

The unique way to do this works, was adding a method calling the parent class in Example class, so:

```
class Example extends Greeting
{
    public hello($name)
    {
        parent::hello($name);
    }
}
```

But it really doesn't a solution, because the idea of extending classes is do not repeat code.
Can you tell if it fixable?

Best Regards.

**Comment by lisachenko** [comment]
Hi, Erik!

Yeap, this issue is one of the known limitation of framework. During the class loading process I don't know information about parent class, because it's not loaded into memory, so I can not override parent method in the child with custom logic. AOP intercepts only concrete methods in the concrete class and can not check methods in extending classes. 

However, it will be possible in the future to recursively load classes and interfaces before PHP to build an hierarchy, but I don't want to implement this logic right now, because it can break a lot of things.

**Comment by eriknyk** [comment]
It would be more usable your framework, because  extensive projects have a lot of extending classes, that is my case,.. I can't use go-aop-php without this capability,..
I think that getting all extending methods, functions and properties recursively should works..
Well, I've waiting for news,.. 
Thanks a lot for your response.

Best Regards.

**Comment by lisachenko** [comment]
@eriknyk you could try to use `feature/static-inheritance` branch, it should handle inheritance now )

**Comment by lisachenko** [comment]
Resolved by #150
